src/main/java/net/sf/jabref/logic/util/version/VersionInformation.java

```diff
@@ -0,0 +1,184 @@
+import java.time.LocalDateTime;
+
+import javax.swing.JOptionPane;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefGUI;
+import net.sf.jabref.JabRefPreferences;
+import net.sf.jabref.gui.help.NewVersionDialog;
+import net.sf.jabref.logic.l10n.Localization;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONObject;
+
+public class VersionInformation {
+
+    private static final Log LOGGER = LogFactory.getLog(VersionInformation.class);
+
+    private final static String JABREF_GITHUB_URL = "https://api.github.com/repos/JabRef/jabref/releases/latest";
+    public final static String JABREF_DOWNLOAD_URL = "http://www.fosshub.com/JabRef.html";
+
+    private Version latestVersion;
+
+
+    /**
+     * this method is meant for a background versions check as it respects if the user previously ignored an update or if it has already been checked within this JabRef instance
+     * The check is started within a Thread and will show a popup after some time if a new Version is available and update the preferecesDialog
+     */
+    public void checkForUpdatesAutomatically() {
+        checkForUpdates(false, true, false);
+    }
+
+    /**
+     * This method should be used when the user explicitly searches for an update as it ignores if the user previuosly ignored an update or if it has already been checked in this JabRef instance
+     * The check is started within a Thread and will show a popup after some time if a new Version is available and update the preferecesDialog
+     */
+    public void checkForUpdatesManually() {
+        checkForUpdates(true, false, true);
+    }
+
+    private void checkForUpdates(boolean forceCheck, boolean respectIgnore, boolean manualExecution) {
+        new Thread() {
+            @Override
+            public void run() {
+                boolean newerVersion = isNewerVersionAvailable(forceCheck, respectIgnore, manualExecution);
+                if (newerVersion) {
+                    new NewVersionDialog();
+                }
+                else if (manualExecution){
+                    JOptionPane.showMessageDialog(JabRefGUI.getMainFrame(),
+                            Localization.lang("uptodate"),
+                            Localization.lang("uptodate"),
+                            JOptionPane.INFORMATION_MESSAGE);
+                }
+                JabRefGUI.getMainFrame().updatePreferenceDialog();
+            }
+        }.start();
+    }
+
+    /**
+     * Use this method for the background version Check
+     * checks if an update is available if enough time has passed since the last check
+     */
+    public void checkIfNecessary() {
+        LocalDateTime lastCheckedDate = LocalDateTime.parse(Globals.prefs.get(JabRefPreferences.VERSION_LAST_CHECKED_DATE));
+        LocalDateTime currentDate = LocalDateTime.now();
+
+        switch (VersionCheckMode.valueOf(Globals.prefs.get(JabRefPreferences.VERSION_CHECK_EVERY))) {
```

Comment:
In the GUI you do the following:
- get current version
- use swing worker
  - (outside EDT) use a logic class to get the latest version, you can use another class which knows whether the latest version is the current version as the current version can be passed to the swing worker
  - (inside EDT) update UI by showing the user something and updating the preferences